graylog2-web-interface/src/views/components/searchbar/saved-search/SavedSearchControls.tsx

```diff
@@ -40,98 +38,57 @@ import type User from 'logic/users/User';
 import ViewPropertiesModal from 'views/components/views/ViewPropertiesModal';
 import { loadAsDashboard, loadNewSearch } from 'views/logic/views/Actions';
 import IfPermitted from 'components/common/IfPermitted';
+import usePluginEntities from 'views/logic/usePluginEntities';
+import {
+  executePluggableSearchDuplicationHandler as executePluggableDuplicationHandler,
+} from 'views/logic/views/pluggableSaveViewFormHandler';
 
 import SavedSearchForm from './SavedSearchForm';
 import SavedSearchList from './SavedSearchList';
 
-type Props = {
-  theme: DefaultTheme,
-  viewStoreState: ViewStoreState,
-};
-
-type State = {
-  showForm: boolean,
-  showList: boolean,
-  showExport: boolean,
-  showShareSearch: boolean,
-  showMetadataEdit: boolean,
-  newTitle: string,
-};
-
 const _isAllowedToEdit = (view: View, currentUser: User | undefined | null) => (
   view.owner === currentUser?.username
   || isPermitted(currentUser?.permissions, [ViewsPermissions.View.Edit(view.id)])
 );
 
-class SavedSearchControls extends React.Component<Props, State> {
-  static propTypes = {
-    theme: PropTypes.shape({
-      colors: PropTypes.object,
-    }).isRequired,
-    viewStoreState: PropTypes.object.isRequired,
-  };
-
-  static contextType = ViewLoaderContext;
-
-  formTarget: { current: null | Button };
-
-  constructor(props: Props) {
-    super(props);
-
-    this.formTarget = React.createRef();
-
-    const { viewStoreState } = props;
-    const { view } = viewStoreState;
-
-    this.state = {
-      showMetadataEdit: false,
-      showExport: false,
-      showShareSearch: false,
-      showForm: false,
-      showList: false,
-      newTitle: (view && view.title) || '',
-    };
-  }
-
-  toggleFormModal = () => {
-    const { showForm } = this.state;
-
-    this.setState({ showForm: !showForm });
-  };
-
-  toggleListModal = () => {
-    const { showList } = this.state;
-
-    this.setState({ showList: !showList });
-  };
-
-  toggleExport = () => {
-    const { showExport } = this.state;
-
-    this.setState({ showExport: !showExport });
-  };
-
-  toggleMetadataEdit = () => {
-    const { showMetadataEdit } = this.state;
-
-    this.setState({ showMetadataEdit: !showMetadataEdit });
-  };
-
-  toggleShareSearch = () => {
-    const { showShareSearch } = this.state;
-
-    this.setState({ showShareSearch: !showShareSearch });
-  };
-
-  onChangeTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
-    this.setState({ newTitle: e.target.value });
+const SavedSearchControls = () => {
```

Comment:
Great work turning this component to a functional component. We should probably do a further breakdown of this component into sub-buttons, as there is a lot of mixed state and the component is pretty big for what it does. This could be in the scope of a follow-up PR though.